### ISSUE TITLE
wrong month or dates shown

### ISSUE THREAD

**Issue by bazo** [eternicode/bootstrap-datepicker#1489]
the calendar shows wrong days for the selected month. the days belong to the month after the month selected in the header

eg.: i have selected may 2015 which has 31 days, calendar shows only 10, when i select a date i get a day in june.

this really sucks

**Comment by bazo** [comment]
problem is on line 918 
orignal this.picker.find('.datepicker-days thead .datepicker-switch')
						.text(DPGlobal.formatDate(new Date(year, month), titleFormat, this.o.language));

should be 
this.picker.find('.datepicker-days thead .datepicker-switch')
						.text(DPGlobal.formatDate(new Date(year, month + 1), titleFormat, this.o.language));

as utc month returns zero indexed month numbers, so in july the month is 6

and i dont understand why you construct another date object when you already have one

**Comment by keibii** [comment]
I've spent the better part of a working day attempting to figure out why the incorrect month label was appearing and have come to the same solution as bazo has suggested above.

**Comment by throup** [comment]
I have encountered the opposite issue; I'm in the UK and I discovered the previous (not next) month was being displayed whenever daylight saving time applied.

That is, for months between October and March, the correct month name was displayed. For other months, the wrong name was displayed (eg in July, the datepicker showed 31 days labelled as "June").

I agree changing line 918 appears to fix the issue, but I would suggest:

```
917 			this.picker.find('.datepicker-days thead .datepicker-switch')
918						.text(DPGlobal.formatDate(new UTCDate(year, month), titleFormat, this.o.language));
```

**Comment by Ephphatha** [comment]
This looks like an inconsistency between the way you build the date and then retrieve components in formatDate(). I think @throup's suggestion of using new UTCDate() would be best since in formatDate you use getUTC*() to get the components.

It seems to be an underflow due to converting from positive local timezone offsets to UTC time.
For example: new Date(2015, 7) gives me a date of Sat Aug 01 2015 00:00:00 GMT+1000. The various getUTC* functions subtract 10 hours from that time so getUTCMonth() returns 6, getUTCDate() returns 31, getUTCHours() returns 14.
new UTCDate(2015, 7) gives me a date of Sat Aug 01 2015 10:00:00 GMT+1000. Subtracting the offset will still yield the same day (with a time of 00:00:00) so the getUTC functions return the expected month, day, hours of 7, 1, 0.

**Comment by acrobat** [comment]
Guys PR #1503 was merged in master, can you verify that this fixes the issue? Thanks!

**Comment by Ephphatha** [comment]
I'm not able to test the latest version from github but we've been using the same modification applied to a version retrieved around 2015/07/20 without further issue for the last quarter.

**Comment by vsn4ik** [comment]
This issue already fixed.

**Comment by acrobat** [comment]
Fixed in #1503
